BackEndFlask/models/utility.py

```diff
@@ -0,0 +1,10 @@
+from core import db
+from sqlalchemy.exc import SQLAlchemyError
+from models.schemas import Course, UserCourse
+
+def get_courses_by_user_courses_by_user_id(user_id):
```

Comment:
Why do need this and get_user_courses_by_user_id_and_role_id? It seems like this serves the same purpose and gives all course information rather than just the id. Is there a time you would just want the course ids?

Reply:
Before I was using get_user_courses_by_user_id_and_role_id() in the controller to later do the joins in the controller. However, I made get_courses_by_user_courses_by_user_id() so that the controller does not need to do any joining itself and sqlachlemy handles the joins for us. We no longer need get_user_courses_by_user_id_and_role_id() so I am going to delete it! I do not believe there is a current situation where we need to only return course ids for the Frontend because all resources retrieved are for purposes of displaying.